packages/commons/src/config-types/workspace-definition/substituteEnvVariables.ts

```diff
@@ -0,0 +1,24 @@
+import { isPlainObject, set } from "lodash";
+
+export function substituteEnvVariables(yaml: unknown): unknown {
```

Comment:
By the time we’re in this file, the yaml has been parsed into a JSON object (by yaml.load), so I wouldn’t expect to see the word yaml anywhere. (Having a bit of trouble following what yaml/processedYaml/target/value are)

Reply:
that makes sense -- stole parts of this code from some other library so can fix it up and make it more sensible